docs/proposals/csi.md

```diff
@@ -0,0 +1,349 @@
+---
+title: Container Storage Interface Proposal
+authors:
+  - "@edisonxiang"
+approvers:
+  - "@qizha"
+  - "@Baoqiang-Zhang"
+  - "@kevin-wangzefeng"
+  - "@m1093782566"
+creation-date: 2019-05-30
+last-updated: 2019-05-30
+status: Pending
+---
+
+# Container Storage Interface Proposal
+
+* [Container Storage Interface Proposal](#container-storage-Interface-proposal)
+  * [Motivation](#motivation)
+  * [Goals](#goals)
+  * [Non\-goals](#non-goals)
+  * [Proposal](#proposal)
+    * [Requirement](#requirement)
+    * [Architecture](#architecture)
+    * [Workflow](#workflow)
+    * [Deployment](#deployment)
+    * [Example](#example)
+  * [Work Items](#work-items)
+
+## Motivation
+
+Currently KubeEdge only supports the following volumes which are based on Kubernetes.
+
+* [emptyDir](https://kubernetes.io/docs/concepts/storage/volumes/#emptydir)
+* [hostPath](https://kubernetes.io/docs/concepts/storage/volumes/#hostpath)
+* [configMap](https://kubernetes.io/docs/concepts/storage/volumes/#configmap)
+* [secret](https://kubernetes.io/docs/concepts/storage/volumes/#secret)
+
+That is not enough for the customers who are using KubeEdge.
+The customers often need to store data and run stateful applications at edge.
+For example, in a system which collects video data and then makes analysis,
+one application stores the video data into a shared storage,
+and another application reads the data from the storage to make analysis.
+KubeEdge should allow customers to store data using StorageClass (SC),
+PersistentVolume (PV) and PersistentVolumeClaim (PVC)
+so that the customers can deploy stateful applications at edge.
```

Comment:
Since there are different types of PV's : https://kubernetes.io/docs/concepts/storage/persistent-volumes/#types-of-persistent-volumes, which ones do we specifically support on the edge ?
Not all can be supported right ?

Reply:
No, It depends on the customers requirement:)
It is maked here a monment ago.
https://github.com/kubeedge/kubeedge/pull/616/files#diff-84e04dec4b2eb39b2f077e5c17ad1b68R97